lapce-proxy/src/lsp.rs

```diff
@@ -84,15 +85,51 @@ impl LspCatalog {
         language_id: &str,
         options: Option<Value>,
     ) {
+        let binary_args = self.get_plugin_binary_args(options.clone());
         let client = LspClient::new(
             language_id.to_string(),
             exec_path,
             options,
+            binary_args.to_owned(),
             self.dispatcher.clone().unwrap(),
         );
         self.clients.insert(language_id.to_string(), client);
     }
 
+    fn get_plugin_binary_args(&mut self, option: Option<Value>) -> Vec<String> {
+        let mut vals = Vec::new();
+
+        let option = match option {
+            Some(val) => val,
+            None => {
+                return vals;
+            }
+        };
+
+        let binary = match option["binary"].as_object() {
+            Some(binary) => binary,
+            None => return vals,
+        };
+
+        let option = match binary.get("binary_args") {
+            Some(binary_args) => binary_args,
+            None => return vals,
+        };
+
+        let option = if let Some(option) = option.as_array() {
+            option
+        } else {
+            println!("binary_args value should be of type [String].");
+            return vals;
+        };
+
+        for val in option {
+            vals.push(String::from(val.as_str().unwrap()));
```

Comment:
This might be much nicer as `option.iter().map(Value::as_str).map(Option::unwrap).(ToString::to_string).collect::<Vec<_>>()` or something like that, but my main issue is the `String::from` here, which just smells weird on a `str`.

You might not even need to specify the type in `collect`, I don't know.

Reply:
Or even

```
let vals = option.iter().filter_map(Value::as_str).map(ToString::to_string).collect::<Vec<_>>()
```